proposals/extensions.md

```diff
@@ -473,58 +493,252 @@ We modify the [member access rules](https://github.com/dotnet/csharpstandard/blo
+          is evaluated as a function pointer invocation.
+        - If it is a value of a type `dynamic`, the *invocation_expression* 
+          is evaluated as a dynamic member invocation.
+      - If the set contains only methods, we remove all the methods that are not 
+        accessible or applicable (see "method invocations").
+        - If the set is empty, proceed to extension methods below.
+        - Otherwise, overload resolution is applied to the candidate methods:
+          - If a single best method is found, the *invocation_expression* 
+            is evaluated as the invocation of this method.
+          - If no single best method is found, a compile-time error occurs.
+
+  - Next, try extension methods (only for the `expr` case):
+    - Check which extension methods in the current scope are eligible.
+      - If the set is empty, proceed to the next enclosing scope.
+      - Otherwise, overload resolution is applied to the candidate set. 
+        - If a single best method is found, the *invocation_expression* 
+          is evaluated as a static method invocation.
+        - If no single best method is found, a compile-time error occurs.
+
+  - Proceed to the next enclosing scope
+- If no extension type member or extension method is found to be suitable for the invocation 
+  in any enclosing scope, a compile-time error occurs.
+
+The preceding rules mean:
+- that instance methods take precedence over extension methods, 
+- that extension type members available in a given namespace take precedence 
+  over extension methods in that namespace,
+- that extension type members available in inner namespace declarations take precedence
+  over extension type members available in outer namespace declarations,
+- that extension methods available in inner namespace declarations take precedence
+  over extension methods available in outer namespace declarations, 
+- that extension type members declared directly in a namespace take precedence
+  over extension type members imported into that same namespace with a using namespace directive,
+- and that extension methods declared directly in a namespace take precedence
+  over extension methods imported into that same namespace with a using namespace directive.
+
+TODO clarify behavior for extension on `object` or `dynamic` used as `dynamic.M()`?
+
+### Indexer access
+
+TL;DR: If no candidate is applicable, then we attempt extension indexer access instead.
+
+We modify the [element access rules](https://github.com/dotnet/csharpstandard/blob/draft-v7/standard/expressions.md#128111-general) as follows:
+
+/[...]
+An *element_access* is dynamically bound if \[...]
+If the *primary_no_array_creation_expression* of an *element_access* is a value of an *array_type*, the *element_access* is an array access. 
+Otherwise, the *primary_no_array_creation_expression* shall be a variable or value of a class, struct, or interface type 
```

Comment:
The if/otherwise/otherwise wording doesn't make sense, the first two cases sound like they together already cover everything, should be probably if/else/otherwise.

```suggestion
Else if the *primary_no_array_creation_expression* is a variable or value of a class, struct, or interface type 
```

Reply:
This is existing spec language. I agree it's probably not formulated the best...
The changes I'm making are highlighted with `**`. I just provided this as context to show where the insertion goes.

Reply:
I know, I think it was formulated fine before you added the second "Otherwise" clause - then I would change the existing "Otherwise" to something else.